src/common/eth-providers/consensus-provider/consensus-provider.service.ts

```diff
@@ -387,79 +393,75 @@ export class ConsensusProviderService {
-      const res = await got
-        .post(urljoin(apiURL, subUrl), { timeout: { response: this.config.get('CL_API_POST_RESPONSE_TIMEOUT') }, ...params })
-        .catch((e) => {
-          if (e.response) {
-            throw new ResponseError(errRequest(e.response.body, subUrl, apiURL), e.response.statusCode);
-          }
-          throw new ResponseError(errCommon(e.message, subUrl, apiURL));
-        });
-      if (res.statusCode !== 200) {
-        throw new ResponseError(errRequest(res.body, subUrl, apiURL), res.statusCode);
-      }
-      try {
-        return JSON.parse(res.body);
-      } catch (e) {
-        throw new ResponseError(`Error converting response body to JSON. Body: ${res.body}`);
-      }
-    });
-  };
-
-  protected apiLargeGet = async (apiURL: string, subUrl: string): Promise<any> => {
-    return await this.prometheus.trackCLRequest(apiURL, subUrl, async () => {
-      return await parseChunked(
-        got.stream
-          .get(urljoin(apiURL, subUrl), { timeout: { response: this.config.get('CL_API_GET_RESPONSE_TIMEOUT') } })
-          .on('response', (r: Response) => {
-            if (r.statusCode != 200) throw new HTTPError(r);
-          }),
-      ).catch((e) => {
-        if (e instanceof HTTPError) {
-          throw new ResponseError(errRequest(<string>e.response.body, subUrl, apiURL), e.response.statusCode);
+  @TrackCLRequest
+  protected async apiGet<T>(apiURL: string, subUrl: string): Promise<T> {
+    const res = await got
+      .get(urljoin(apiURL, subUrl), { timeout: { response: this.config.get('CL_API_GET_RESPONSE_TIMEOUT') } })
+      .catch((e) => {
+        if (e.response) {
+          throw new ResponseError(errRequest(e.response.body, subUrl, apiURL), e.response.statusCode);
         }
         throw new ResponseError(errCommon(e.message, subUrl, apiURL));
       });
-    });
-  };
+    if (res.statusCode !== 200) {
+      throw new ResponseError(errRequest(res.body, subUrl, apiURL), res.statusCode);
+    }
+    try {
+      return JSON.parse(res.body);
+    } catch (e) {
+      throw new ResponseError(`Error converting response body to JSON. Body: ${res.body}`);
+    }
+  }
 
-  protected apiLargePost = async (apiURL: string, subUrl: string, params?: Record<string, any>): Promise<any> => {
-    return await this.prometheus.trackCLRequest(apiURL, subUrl, async () => {
-      return await parseChunked(
-        got.stream
-          .post(urljoin(apiURL, subUrl), { timeout: { response: this.config.get('CL_API_POST_RESPONSE_TIMEOUT') }, ...params })
-          .on('response', (r: Response) => {
-            if (r.statusCode != 200) throw new HTTPError(r);
-          }),
-      ).catch((e) => {
-        if (e instanceof HTTPError) {
-          throw new ResponseError(errRequest(<string>e.response.body, subUrl, apiURL), e.response.statusCode);
+  @TrackCLRequest
+  protected async apiPost<T>(apiURL: string, subUrl: string, params?: Record<string, any>): Promise<T> {
```

Comment:
What advantages of using this method over `apiLargePost` for, _ughm_, small requests?

Reply:
It is expected that simple JS `JSON -> Object` casting for small JSON will be faster than casting in stream used in `apiLargePost`

Reply:
But since it is small, do we really need this method?

Reply:
Currently, it's not used somewhere, but it will (for other endpoints in withdrawals monitoring purposes in the future)

Reply:
I still don't see enough motivation to keep the method.

Reply:
done